src/wc.js

```diff
@@ -0,0 +1,159 @@
+/*  
+    copy tail.sh as an example, and provide a similar function to wc.
+    wc man page:http://man7.org/linux/man-pages/man1/wc.1.html
+    (notes from man page)        
+       wc - Print newline, word, and byte counts for each FILE, and a total line
+       if more than one FILE is specified.  A word is a non-zero-length
+       sequence of characters delimited by white space.
+        -c, --bytes print the byte counts
+        -m, --chars print the character counts
+        -l, --lines print the newline counts
+        --files0-from=F
+                    read input from the files specified by NUL-terminated names in
+                    file F; If F is - then read names from standard input
+        -L, --max-line-length print the maximum display width
+        -w, --words print the word counts
+        --help display this help and exit
+        --version output version information and exit
+*/
+
+var common = require('./common');
+var fs = require('fs');
+
+common.register('wc', _wc, {
+    canReceivePipe: true,
+    cmdOptions: {
+        'm': 'charCount',
+        'l': 'lineCount',
+        'w': 'wordCount',
+    },
+});
+
+function getLines(data) {
+    //data is the file contents
+    var lines = data.split('\n'); // array
+
+    if (lines[lines.length - 1] === '') {
+        //in case it wanted to count an empty line at the bottom
+        lines.pop();
+    }
+
+    return lines.length.toString();
+}
+
+function getWords(data) {
+    var regexSplit = /\s/;
+    var wordsInitial = data.split(regexSplit); //array
+
+    while (wordsInitial.indexOf('') !== -1) {
+        //with words, cut out any blanks enumerated
+        wordsInitial.splice(wordsInitial.indexOf(''), 1);
+    }
+    return wordsInitial.length.toString();
+}
+
+function getChars(data) {
+    return data.length.toString();
+}
+
+//@
+//@ ### wc([options,] file [, file ...])
+//@ ### wc([options,] file_array)
+//@ Available options:
+//@ + `-m`: print the char counts 
+//@ + `-l`: print the newline counts
+//@ + `-w`: print the word counts (space delimited characters)
+//@ 
+//@ Examples:
+//@
+//@ ```javascript
+//@ var counts = wc('file.txt'); //returns charCount lineCount wordCount
+//@ var charCount = wc('-m','file.txt'); //returns number of characters
+//@ var lineCount = wc('-l','file.txt'); //returns number of newlines
+//@ var wordCount = wc('-w','file.txt'); //returns number of words 
+//@ var countFiles = wc('file1', 'file2'); // returns: lineCount wordCount charCount 'file1'\nlineCount wordCount charCount 'file2' \n totalLines totalWords totalChars 'total'
```

Comment:
Column limit. Maybe we can move this onto its own line? Or we could just remove this example.

Reply:
i'll probably remove that example.